xds/internal/xdsclient/client_new.go

```diff
@@ -84,38 +88,25 @@ func newWithConfig(config *bootstrap.Config, watchExpiryTimeout time.Duration, i
 	return c, nil
 }
 
-// NewWithConfigForTesting returns an xDS client for the specified bootstrap
-// config, separate from the global singleton.
-//
-// The second return value represents a close function which the caller is
-// expected to invoke once they are done using the client.  It is safe for the
-// caller to invoke this close function multiple times.
-//
-// # Testing Only
-//
-// This function should ONLY be used for testing purposes.
-// TODO(easwars): Document the new close func.
-func NewWithConfigForTesting(config *bootstrap.Config, watchExpiryTimeout, authorityIdleTimeout time.Duration) (XDSClient, func(), error) {
-	cl, err := newWithConfig(config, watchExpiryTimeout, authorityIdleTimeout)
-	if err != nil {
-		return nil, nil, err
-	}
-	return cl, grpcsync.OnceFunc(cl.close), nil
-}
-
-func init() {
-	internal.TriggerXDSResourceNameNotFoundClient = triggerXDSResourceNameNotFoundClient
+// OptionsForTesting contains options to configure xDS client creation for
+// testing purposes only.
+type OptionsForTesting struct {
+	// Contents contain a JSON representation of the bootstrap configuration to
+	// be used when creating the xDS client.
+	//
+	// Config field takes precedence over the Contents field.
```

Comment:
mm it's not clear to me which `Config` field  does this ref

Reply:
Thanks. I initially had a `Config` and a `Contents`, but figured that there was no need to give two ways of doing the same thing to tests. So, I had gotten rid of `Contents`, but forgot to update this comment.